### ISSUE TITLE
ITenantIdentificationStrategy Cannot Access Authenticated User in Core 3.x

### ISSUE THREAD

**Issue by jdroman2302** [autofac/Autofac.AspNetCore.Multitenant#23]
Hi, this may not be an issue per se, but a question about whether something is possible with Autofac multitenant components for asp.net core and JWT authentication.  I have some middleware setup in my Asp.Net Core 3.1 app (.AddJwtBearer(), followed by .UseAuthentication() later).  This produces an authenticated user on the HttpContext.User.Identity, both in controller code and in Startup middleware code.  However, I am unable to access this from my ITenantIdentificationStrategy implementation, which looks at IHttpContextAccessor's User property.  When it hits the strategy class code, the User.Identity.IsAuthenticated is false and there are no claims.

My question, is this scenario even supported?  Or does Autofac call the tenant identifier component before the Jwt token middleware has a chance to run?

I have an alternate solution, so it's not a huge deal, but I'd prefer to use the built-in User structures during tenant resolution if possible.

If it should be possible and you are willing to help me troubleshoot, I will be happy to provide a minimal repro set.

Thanks!

**Comment by tillig** [comment]
The tenant lifetime scope is determined pretty early in the pipeline. It needs to be because things like action filters and so on all need a lifetime scope, and all of that is potentially tenant-specific. It effectively replaces the same point when standard `HttpContext.RequestServices` would normally be chosen, which, again, is super early in the pipeline. That means `ITenantIdentificationStrategy` is going to run very early.

I've found in my apps before that I've had to do some very minor duplication of effort to manually verify that a token is good and then investigate claims in the token myself (caching that result in `HttpContext.Items`) because sometimes the actual auth runs later. If I can verify _just enough_ to ensure I've got the right tenant, the rest of the token validation can come later.

Unfortunately, I don't have time to help troubleshoot. I barely have time to address the issues and pull requests on core Autofac and the other 20+ integration libraries, even with the help we've gotten. You're going to have to troubleshoot that yourself.
